src/content/reference/react/use-client.md

```diff
@@ -23,37 +23,364 @@ canary: true
+* `'use client'` must be at the very beginning of a file, above any imports or other code (comments are OK). They must be written with single or double quotes, but not backticks.
+* When a `'use client'` module is imported from another client-rendered module, the directive has no effect.
+* When a component module contains a `'use client'` directive, any usage of that component is guaranteed to be a Client Component. However, a component can still be evaluated on the client even if it does not have a `'use client'` directive.
+	* A component usage is considered a Client Component if it is defined in module with `'use client'` directive or it is a transitive dependency of a module that contains a `'use client'` directive. Otherwise, it is a Server Component.
+* Code that is marked for client evaluation is not limited to components. All code that is a part of the client module sub-tree is sent to and run by the client.
+* When a server evaluated module imports values from a `'use client'` module, the values must either be a React component or [supported serializable prop values](#passing-props-from-server-to-client-components) to be passed to a Client Component. Any other use case will throw an exception.
+
 
 ## Usage {/*usage*/}
 
-<Wip>
-This section is a work in progress. 
+### Building with interactivity and state {/*building-with-interactivity-and-state*/}
+
+<Sandpack>
+
+```js App.js
+'use client';
+
+import {useState} from 'react';
+
+export default function Counter({initialValue = 0}) {
+	const [countValue, setCountValue] = useState(initialValue);
+	const increment = () => setCountValue(countValue + 1);
+	const decrement = () => setCountValue(countValue - 1);
+	return (<>
+		<h2>Count Value: {countValue}</h2>
+		<button onClick={increment}>+1</button>
+		<button onClick={decrement}>-1</button>
+	</>);
+}
+```
+
+</Sandpack>
+
+As `Counter` requires both the `useState` hook and event handlers to increment or decrement the value, this component must be a Client Component and will require a `'use client'` directive at the top.
+
+In contrast, a component that renders UI without interaction will not need to be a Client Component.
+
+```js
+import {getCounterValueFromFile} from 'fs-utils';
+import Counter from './Counter';
+
+export default async function CounterContainer() {
+  const initialValue = await getCounterValueFromFile();
+  return <Counter initialValue={initialValue} />
+}
+```
+
+For example, the parent component of the Client Component `CounterContainer` does not require `'use client'` as it is not interactive and does not use state. To be more specific, `CounterContainer` must be a Server Component as it reads from the local file system on the server. Notably, Server Components support asynchronous rendering, which is not supported for Client Components.
```

Comment:
To be more specific, -->  In addition, / Moreover,  —— this is just a further fact rather than an increase of specificity.

Notably, Server Components support asynchronous rendering, which is not supported for Client Components. ---> (This is possible because Server Components, unlike Client Components, can be async functions.)

I would say "can be async functions" because this is more specific and actionable than "support asynchronous rendering" which is an abstract concept that could refer to concurrent rendering or who knows whatw.